app/src/main/java/nextstep/signup/MainActivity.kt

```diff
@@ -3,44 +3,29 @@ package nextstep.signup
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material3.MaterialTheme
-import androidx.compose.material3.Surface
-import androidx.compose.material3.Text
-import androidx.compose.runtime.Composable
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Scaffold
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.tooling.preview.Preview
+import nextstep.signup.ui.signup.SignUpScreen
 import nextstep.signup.ui.theme.SignupTheme
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             SignupTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colorScheme.background
-                ) {
-                    Greeting("Android")
+                Scaffold { innerPadding ->
+                    Box(
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .padding(innerPadding),
+                    ) {
+                        SignUpScreen()
+                    }
```

Comment:
저는 개인적으로 setContent내의 컴포져블 함수를 최소화 하는편입니다!
지회님은 컴포져블 함수를 보통 어떤 기준으로 나누시는 편인가요? 😄

Reply:
1. setContent에서 가장 최상위단에 `Scaffold`를 많이 이용하는 편입니다. setContent 내 컴포져블을 최소화 하기 위해서 `SignUpApp` 함수를 만들어서 처리해도 될까요?
```suggestion
                SignUpApp()
```

```
@Composable
private fun SignUpApp() {
    Scaffold { innerPadding ->
        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding),
        ) {
            SignUpScreen()
        }
    }
}
```


2. 컴포져블 함수를 나눌 때의 기준은 예를들어 step2 과제중에 Password용 TextField인 PasswordTextField, Email용 TextField인 EmailTextField를 만드는 것을 기준으로 잡지 않았습니다. 제가 생각하기에 좀 더 범용적으로 사용할 수 있는 부분은 `TextField`의 `colors`를 재사용하는 것이 더 큰 이점이라 생각하여 `NSTextField`를 생성하게 되었습니다.